services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/FlowValidationNegativeSpec.groovy

```diff
@@ -0,0 +1,205 @@
+package org.openkilda.functionaltests.spec.northbound.flows
+
+import groovy.util.logging.Slf4j
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.PathHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslInfoData
+import org.openkilda.messaging.model.Flow
+import org.openkilda.messaging.model.SwitchId
+import org.openkilda.northbound.dto.flows.FlowValidationDto
+import org.openkilda.testing.Constants
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.database.Database
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+
+@Slf4j
+@Narrative("""The specification covers the following scenarios:
+              -- Deleting flow rule from a switch and check if switch and flow validation fails.
+              -- Failed switch validation should not cause validation errors for flows with all rules in place.
+              Test case permutations (full-factored):
+                 - forward and reverse flows
+                 - ingress, transit and egress switches
+                 - Single switch, two switch and three+ switch flow spans.
+            """)
+class FlowValidationNegativeSpec extends BaseSpecification {
+
+    @Autowired
+    PathHelper pathHelper
+
+    @Autowired
+    Database database
+
+    /**
+     * Checks if there is no discrepancies in the flow validation results
+     * //TODO: Don't skip MeterId discrepancies when virtual env starts supporting the meters.
+     * @param flow Flow validation results
+     * @return boolean
+     */
+    boolean isFlowValid(FlowValidationDto flow) {
```

Comment:
Please, move all helper methods under tests. It is inconvenient to scroll down a lot to find the test.